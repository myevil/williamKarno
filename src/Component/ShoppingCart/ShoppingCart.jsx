import React, { Component } from 'react'

class ShoppingCart extends Component {

  handleChangeQuantity(e, changeProductCart) {
    if(e.target.value >= 0){
      changeProductCart.quantity = e.target.value;
      changeProductCart.total = changeProductCart.quantity * changeProductCart.price;
      this.props.handleChangeQuantity(changeProductCart);
    }
  }

  handleDeleteProductOnCart(deletedProduct){
    this.props.handleDeleteProduct(deletedProduct);
  }

  render() {
    const {ShoppingCart} = this.props
    return (
      <React.Fragment>
        {ShoppingCart.length !== 0 ? ShoppingCart.map(CartData => {
          return (
            <tr key={CartData.name}>
              <td>
                <button onClick={() => this.handleDeleteProductOnCart(CartData)}>close</button>
              </td>
              <td>
                <img src={CartData.image} width='50px' height='50' alt='product'/>
                <span>{CartData.name}</span>
              </td>
              <td>{CartData.price}</td>
              <td>
                <input
                  type='number'
                  value={CartData.quantity}
                  onChange={e => this.handleChangeQuantity(e, CartData)}
                />
              </td>
              <td>{CartData.total}</td>
            </tr>
          )
        }) : <td>no data</td>}
      </React.Fragment>
    )
  }
}

export default ShoppingCart
