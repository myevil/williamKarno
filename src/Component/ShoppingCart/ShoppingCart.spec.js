import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShoppingCart from './ShoppingCart';

configure({ adapter: new Adapter() });

describe('test hompage', () => {
  const wrapper = shallow(<ShoppingCart
    ShoppingCart={[]}
  />)
  it('handleChangeQuantity should working properly');
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
