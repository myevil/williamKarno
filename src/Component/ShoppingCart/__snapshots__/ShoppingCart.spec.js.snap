// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hompage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShoppingCart
    ShoppingCart={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <td>
        no data
      </td>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "no data",
      },
      "ref": null,
      "rendered": "no data",
      "type": "td",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <td>
          no data
        </td>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "no data",
        },
        "ref": null,
        "rendered": "no data",
        "type": "td",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
